dbms/src/Interpreters/ExpressionActions.h

```diff
@@ -230,7 +230,9 @@ struct ExpressionActionsChain
     struct Step
     {
         ExpressionActionsPtr actions;
+        NameSet additional_input;
         Names required_output;
+        std::vector<std::shared_ptr<bool>> can_remove_required_output; /// Has the same size with required_output, is filled in finalize()
```

Comment:
If the output is required, why it's possible that we "can remove" it? No explanation.
Non obvious.